pkg/decision/evaluator/matchers/semver.go

```diff
@@ -0,0 +1,207 @@
+/****************************************************************************
+ * Copyright 2020, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package matchers //
+package matchers
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/optimizely/go-sdk/pkg/decision/reasons"
+	"github.com/optimizely/go-sdk/pkg/entities"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	buildSeperator      = "+"
+	preReleaseSeperator = "-"
+	whiteSpace          = " "
+)
+
+var digitCheck = regexp.MustCompile(`^[0-9]+$`)
+
+// SemanticVersion defines the class
+type SemanticVersion struct {
+	Condition string // condition is always a string here
+}
+
+func (sv SemanticVersion) compareVersion(attribute string) (int, error) {
+
+	targetedVersionParts, err := sv.splitSemanticVersion(sv.Condition)
+	if err != nil {
+		return 0, err
+	}
+	versionParts, e := sv.splitSemanticVersion(attribute)
+	if e != nil {
+		return 0, e
+	}
+
+	// Up to the precision of targetedVersion, expect version to match exactly.
+	for idx := range targetedVersionParts {
+
+		switch {
+		case len(versionParts) <= idx:
+			return -1, nil
```

Comment:
here you want to pass 1 if targetversion is a prerelease. otherwise -1.

sv.isPreRelease() ? 1  : -1